cluster/v1alpha1/types_rolloutstrategy.go

```diff
@@ -0,0 +1,96 @@
+package v1alpha1
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:deepcopy-gen=true
+
+// RolloutStrategy API used by consumer workload applier APIs such as ManifestWorkReplicaSet, Policy, ManagedClusterAddon ...etc
+// to define how the workload will be applied to the selected clusters. For more information check the placementStrategy
+// proposal at https://github.com/open-cluster-management-io/enhancements/tree/main/enhancements/sig-architecture/64-placementStrategy
+
+// Rollout strategy to be used by workload applier controller.
+type RolloutStrategy struct {
+	// Rollout strategy Types are All, Progressive and ProgressivePerGroup
+	// 1) All means apply the workload to all clusters in the decision groups at once.
+	// 2) Progressive means apply the workload to the selected clusters progressively per cluster. The workload will not be applied to the next cluster unless one of the current applied clusters reach the successful state or timeout.
+	// 3) ProgressivePerGroup means apply the workload to decisionGroup clusters progressively per group. The workload will not be applied to the next decisionGroup unless all clusters in the current group reach the successful state or timeout.
+	// +kubebuilder:default:=All
+	// +optional
```

Comment:
this is need to an enum validation

Reply:
Still need some const  string like after adding enum validation.

```
const (
	//All means apply the workload to all clusters in the decision groups at once.
	All string = "All"
	//Progressive means apply the workload to the selected clusters progressively per cluster.
	Progressive string = "Progressive"
	//ProgressivePerGroup means apply the workload to the selected clusters progressively per group.
	ProgressivePerGroup string = "ProgressivePerGroup"
)
```

Reply:
ops, removed by mistake. done